corehq/apps/locations/resources/v0_1.py

```diff
@@ -20,13 +24,22 @@ def obj_get(self, bundle, **kwargs):
 
     def obj_get_list(self, bundle, **kwargs):
         domain = kwargs['domain']
+        project = bundle.request.project
         parent_id = bundle.request.GET.get('parent_id', None)
         include_inactive = json.loads(bundle.request.GET.get('include_inactive', 'false'))
-        if parent_id:
+        user = bundle.request.couch_user
+        viewable = viewable_locations_ids(user, project)
+
+        if not parent_id:
+            locs = root_locations(domain)
+        else:
             parent = get_object_or_not_exist(Location, parent_id, domain)
-            return parent.sql_location.child_locations(include_archive_ancestors=include_inactive)
+            locs = parent.sql_location.child_locations(include_archive_ancestors=include_inactive)
+
+        return [child for child in locs if child.location_id in viewable]
 
-        return root_locations(domain)
+    def dehydrate_can_edit(self, bundle):
```

Comment:
where is this used? 
small: also not sure what `dehydrate` means in this context, maybe a comment?